mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowTrackingContext.java

```diff
@@ -0,0 +1,59 @@
+package org.mlflow.tracking;
+
+import org.mlflow.api.proto.Service.*;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+public class MlflowTrackingContext {
+
+    private MlflowClient client;
+    private String experimentId;
+    private Deque<ActiveRun> activeRunStack = new ArrayDeque<>();
+
+    public MlflowTrackingContext() {
+        this(new MlflowClient(), MlflowClient.DEFAULT_EXPERIMENT_ID);
+    }
+
+    public MlflowTrackingContext(MlflowClient client) {
```

Comment:
Can we have some constructors that don't take an MLflowClient but rather take the most common params for that? It's kind of weird to make people learn 2 classes just to initialize this higher level API. Maybe one advanced constructor can take a client, but there should be easy ones that don't (the default constructor is good, but maybe add some that take a tracking URI and the credentials provider thing for other common use cases).